components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java

```diff
@@ -269,6 +269,149 @@ public Response getDevices(
         }
     }
 
+    @GET
+    @Path("/v2")
+    @Deprecated
+    @Override
+    public Response getDevicesDep(
+            @QueryParam("name") String name,
+            @QueryParam("type") String type,
+            @QueryParam("user") String user,
+            @QueryParam("userPattern") String userPattern,
+            @QueryParam("role") String role,
+            @QueryParam("ownership") String ownership,
+            @QueryParam("status") String status,
+            @QueryParam("groupId") int groupId,
+            @QueryParam("since") String since,
+            @HeaderParam("If-Modified-Since") String ifModifiedSince,
+            @QueryParam("offset") int offset,
+            @QueryParam("limit") int limit) {
+        try {
+            if (!StringUtils.isEmpty(name) && !StringUtils.isEmpty(role)) {
+                return Response.status(Response.Status.BAD_REQUEST).entity(
+                        new ErrorResponse.ErrorResponseBuilder().setMessage("Request contains both name and role " +
+                                "parameters. Only one is allowed " +
+                                "at once.").build()).build();
+            }
+            RequestValidationUtil.validatePaginationParameters(offset, limit);
+            DeviceManagementProviderService dms = DeviceMgtAPIUtils.getDeviceManagementService();
+            DeviceAccessAuthorizationService deviceAccessAuthorizationService =
+                    DeviceMgtAPIUtils.getDeviceAccessAuthorizationService();
+            if (deviceAccessAuthorizationService == null) {
+                return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(
+                        new ErrorResponse.ErrorResponseBuilder().setMessage("Device access authorization service is " +
+                                "failed").build()).build();
+            }
+            PaginationRequest request = new PaginationRequest(offset, limit);
+            PaginationResult result;
+            DeviceList devices = new DeviceList();
+
+            if (name != null && !name.isEmpty()) {
+                request.setDeviceName(name);
+            }
+            if (type != null && !type.isEmpty()) {
+                request.setDeviceType(type);
+            }
+            if (ownership != null && !ownership.isEmpty()) {
+                RequestValidationUtil.validateOwnershipType(ownership);
+                request.setOwnership(ownership);
+            }
+            if (status != null && !status.isEmpty()) {
+                RequestValidationUtil.validateStatus(status);
+                request.setStatus(status);
+            }
+            if (groupId != 0) {
+                request.setGroupId(groupId);
+            }
+            if (role != null && !role.isEmpty()) {
+                request.setOwnerRole(role);
+            }
+
+            String authorizedUser = MultitenantUtils.getTenantAwareUsername(CarbonContext.getThreadLocalCarbonContext().getUsername());
```

Comment:
@lasanthaDLPDS this line seems longer than the maximum length allowed. Please fix.

Reply:
This source no longer exists.